src/programs/unblur/unblur.cpp

```diff
@@ -878,16 +915,19 @@ bool UnBlurApp::DoCalculation()
 		delete my_electron_dose;
 		delete [] dose_filter_sum_of_squares;
 	}
-
-	overall_finish = wxDateTime::Now();
-
-	//SendInfo(wxString::Format("Timings for %s: Overall: %s; reading, gain, FT, resampling of stack: %s; initial ali: %s; main ali: %s; unbinned ali: %s\n",input_filename,(overall_finish-overall_start).Format(),(read_frames_finish-read_frames_start).Format(),(first_alignment_finish-first_alignment_start).Format(),(main_alignment_finish-main_alignment_start).Format(),(final_alignment_finish-final_alignment_start).Format()));
+	profile_timing.lap("cleanup");
+	unblur_timing.print_times();
+	profile_timing.print_times();
+	profile_timing_refinement_method.print_times();
 
 	return true;
 }
 
-void unblur_refine_alignment(Image *input_stack, int number_of_images, int max_iterations, float unitless_bfactor, bool mask_central_cross, int width_of_vertical_line, int width_of_horizontal_line, float inner_radius_for_peak_search, float outer_radius_for_peak_search, float max_shift_convergence_threshold, float pixel_size, int number_of_frames_for_running_average, int savitzy_golay_window_size, int max_threads, float *x_shifts, float *y_shifts)
+void unblur_refine_alignment(Image *input_stack, int number_of_images, int max_iterations, float unitless_bfactor, bool mask_central_cross, int width_of_vertical_line, int width_of_horizontal_line, float inner_radius_for_peak_search, float outer_radius_for_peak_search, float max_shift_convergence_threshold, float pixel_size, int number_of_frames_for_running_average, int savitzy_golay_window_size, int max_threads, float *x_shifts, float *y_shifts, StopWatch& profile_timing_refinement_method)
 {
+
+	profile_timing_refinement_method.mark_entry_or_exit_point();
```

Comment:
Why does the profile_timing_refinement need the `mark_entry_or_exit_point` method and the two other timers do not?

Reply:
profile_timing_refinement is only "active" while the refinement method is active, and the refinement is called multiple times during the unblur run. The mark_entry_or_exit_point effectively pauses/restarts the elapsed time, such that it records only the time elapsed while **inside** the refinement method and not the time elapsed since the first time the method was called.